### PR TITLE
Fix `build` action for macOS

### PR DESCRIPTION
**Pull request by Ailrun**
We don't have the macOS version for HLS 8.10.5, which broke the assumption of `build` action for releases.

### CODE REVIEW

**Review by jneira — APPROVED** [review]
lgtm

**Comment by Ailrun** [comment]
Currently benchmark fails for some unknown reason, so I will merge this for sake of the release.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -115,11 +115,11 @@ jobs:
         path: ${{ steps.compress_server_binary.outputs.path }}
 
     - name: Build Wrapper
-      if: matrix.ghc == '8.10.5'
+      if: matrix.ghc == '8.10.5' && matrix.os != 'macOS-latest' || matrix.ghc == '8.10.4' && matrix.os == 'macOS-latest'
```
Comment (berberman) [inline .github/workflows/build.yml:118]: Could we simply use GHC 8.10.4 to build the wrapper for all platforms until the next stable GHC version released?

Comment (Ailrun) [reply .github/workflows/build.yml:118]: Oops, thank you for the review. Maybe that's better as GHC 8.10.5 also does not expose `allocateWrite` symbol (which causes some GHC-plugin failures).

**Pull request closed without merging**